salt/cloud/clouds/digital_ocean.py

```diff
@@ -414,12 +447,23 @@ def __query_node_data(vm_name):
             pass
         finally:
             raise SaltCloudSystemExit(str(exc))
-
-    for network in data['networks']['v4']:
-        if network['type'] == 'public':
-            ip_address = network['ip_address']
+     
+    # add DNS records, set ssh_host, default to first found IP, preferring IPv4 for ssh bootstrap script target
+    addr_families, dns_arec_types = (('v4', 'v6'), ('A', 'AAAA'))
+    arec_map = dict(zip(addr_families, dns_arec_types))
+    for facing, addr_family, ip_address in [(net['type'], family, net['ip_address'])
+                                            for family in addr_families
+                                            for net in data['networks'][family]]:
+        log.info('found {0} IP{1} interface for "{2}"'.format(facing, addr_family, ip_address))
+        dns_rec_type = arec_map[addr_family]
+        if facing == 'public':
+            dnsrv = None
+            if create_dns_record:
+                dnsrv = __add_dns_addr__(dns_rec_type, ip_address)
+            if 'ssh_host' not in vm_ or not vm_['ssh_host']:
+                vm_['ssh_host'] = ip_address
+    
     vm_['key_filename'] = key_filename
-    vm_['ssh_host'] = ip_address
```

Comment:
In most cases, the salt master (or whatever host is running the bootstrap), will only be able to reach the public IP of the new DigitalOcean droplet. AFAIK, *all* DigitalOcean droplets have a public IP. If somehow DigitalOcean fails to provide (or indicate) a public interface IP address, then bootstrap will fail on SSH connection failure anyway. I propose to add a warning message in that (corner) case. The current functionality will crash with an exception "NameError: name 'ip_address' is not defined" in that case. Do we need to preserve that failure mode?